torch/distributed/checkpoint/filesystem.py

```diff
@@ -435,15 +437,18 @@ def finish(
         metadata.storage_data = storage_md
         with (self.path / ".metadata.tmp").open("wb") as metadata_file:
             pickle.dump(metadata, metadata_file)
-            os.fsync(metadata_file.fileno())
+            if self.sync_files:
+                os.fsync(metadata_file.fileno())
 
         (self.path / ".metadata.tmp").rename(self.path / ".metadata")
 
 
 class FileSystemReader(StorageReader):
     def __init__(self, path: Union[str, os.PathLike]) -> None:
         super().__init__()
-        self.path = Path(path)
+        if not isinstance(path, Path):
+            path = Path(path)
+        self.path = path
```

Comment:
nit: 
```
maybe self.path = path if isinstance(path, Path) else Path(path)
```

Reply:
I'm going to keep this unchanged unless pytorch guideline prefers one liner.